From ed2cd372fc200ecb6acc9303fbfe31e5d29c0940 Mon Sep 17 00:00:00 2001
From: Daiane Angolini <daiane.angolini@foundries.io>
Date: Thu, 26 Dec 2024 22:24:23 +0000
Subject: [PATCH] imx:dts:imx8mm-evkb: fix the pmic name to avoid duplicated
 label error

Since commit [1] changes the imx8mm-evk.dtsi node label and does not change the
imx8evkb.dts.

Fix the error:
|   DTC     arch/arm64/boot/dts/freescale/imx8mm-evkb.dtb
.../imx8mm-lpddr4-evk/kernel-source/arch/arm64/boot/dts/freescale/imx8mm-evkb.dts:28.21-35.6: ERROR (duplicate_label): /soc@0/bus@30800000/i2c@30a20000/pmic@25/regulators/BUCK1: Duplicate label 'buck1_reg' on /soc@0/bus@30800000/i2c@30a20000/pmic@25/regulators/BUCK1 and /soc@0/bus@30800000/i2c@30a20000/pca9450@25/regulators/BUCK1
.../imx8mm-lpddr4-evk/kernel-source/arch/arm64/boot/dts/freescale/imx8mm-evkb.dts:38.21-46.6: ERROR (duplicate_label): /soc@0/bus@30800000/i2c@30a20000/pmic@25/regulators/BUCK2: Duplicate label 'buck2_reg' on /soc@0/bus@30800000/i2c@30a20000/pmic@25/regulators/BUCK2 and /soc@0/bus@30800000/i2c@30a20000/pca9450@25/regulators/BUCK2
.../imx8mm-lpddr4-evk/kernel-source/arch/arm64/boot/dts/freescale/imx8mm-evkb.dts:49.21-55.6: ERROR (duplicate_label): /soc@0/bus@30800000/i2c@30a20000/pmic@25/regulators/BUCK3: Duplicate label 'buck3_reg' on /soc@0/bus@30800000/i2c@30a20000/pmic@25/regulators/BUCK3 and /soc@0/bus@30800000/i2c@30a20000/pca9450@25/regulators/BUCK3
.../imx8mm-lpddr4-evk/kernel-source/arch/arm64/boot/dts/freescale/imx8mm-evkb.dts:58.21-64.6: ERROR (duplicate_label): /soc@0/bus@30800000/i2c@30a20000/pmic@25/regulators/BUCK4: Duplicate label 'buck4_reg' on /soc@0/bus@30800000/i2c@30a20000/pmic@25/regulators/BUCK4 and /soc@0/bus@30800000/i2c@30a20000/pca9450@25/regulators/BUCK4
.../imx8mm-lpddr4-evk/kernel-source/arch/arm64/boot/dts/freescale/imx8mm-evkb.dts:67.21-73.6: ERROR (duplicate_label): /soc@0/bus@30800000/i2c@30a20000/pmic@25/regulators/BUCK5: Duplicate label 'buck5_reg' on /soc@0/bus@30800000/i2c@30a20000/pmic@25/regulators/BUCK5 and /soc@0/bus@30800000/i2c@30a20000/pca9450@25/regulators/BUCK5
.../imx8mm-lpddr4-evk/kernel-source/arch/arm64/boot/dts/freescale/imx8mm-evkb.dts:76.21-82.6: ERROR (duplicate_label): /soc@0/bus@30800000/i2c@30a20000/pmic@25/regulators/BUCK6: Duplicate label 'buck6_reg' on /soc@0/bus@30800000/i2c@30a20000/pmic@25/regulators/BUCK6 and /soc@0/bus@30800000/i2c@30a20000/pca9450@25/regulators/BUCK6
.../imx8mm-lpddr4-evk/kernel-source/arch/arm64/boot/dts/freescale/imx8mm-evkb.dts:85.19-91.6: ERROR (duplicate_label): /soc@0/bus@30800000/i2c@30a20000/pmic@25/regulators/LDO1: Duplicate label 'ldo1_reg' on /soc@0/bus@30800000/i2c@30a20000/pmic@25/regulators/LDO1 and /soc@0/bus@30800000/i2c@30a20000/pca9450@25/regulators/LDO1
.../imx8mm-lpddr4-evk/kernel-source/arch/arm64/boot/dts/freescale/imx8mm-evkb.dts:94.19-100.6: ERROR (duplicate_label): /soc@0/bus@30800000/i2c@30a20000/pmic@25/regulators/LDO2: Duplicate label 'ldo2_reg' on /soc@0/bus@30800000/i2c@30a20000/pmic@25/regulators/LDO2 and /soc@0/bus@30800000/i2c@30a20000/pca9450@25/regulators/LDO2
.../imx8mm-lpddr4-evk/kernel-source/arch/arm64/boot/dts/freescale/imx8mm-evkb.dts:103.19-109.6: ERROR (duplicate_label): /soc@0/bus@30800000/i2c@30a20000/pmic@25/regulators/LDO3: Duplicate label 'ldo3_reg' on /soc@0/bus@30800000/i2c@30a20000/pmic@25/regulators/LDO3 and /soc@0/bus@30800000/i2c@30a20000/pca9450@25/regulators/LDO3
.../imx8mm-lpddr4-evk/kernel-source/arch/arm64/boot/dts/freescale/imx8mm-evkb.dts:112.19-118.6: ERROR (duplicate_label): /soc@0/bus@30800000/i2c@30a20000/pmic@25/regulators/LDO4: Duplicate label 'ldo4_reg' on /soc@0/bus@30800000/i2c@30a20000/pmic@25/regulators/LDO4 and /soc@0/bus@30800000/i2c@30a20000/pca9450@25/regulators/LDO4
.../imx8mm-lpddr4-evk/kernel-source/arch/arm64/boot/dts/freescale/imx8mm-evkb.dts:121.19-125.6: ERROR (duplicate_label): /soc@0/bus@30800000/i2c@30a20000/pmic@25/regulators/LDO5: Duplicate label 'ldo5_reg' on /soc@0/bus@30800000/i2c@30a20000/pmic@25/regulators/LDO5 and /soc@0/bus@30800000/i2c@30a20000/pca9450@25/regulators/LDO5

[1] https://github.com/Freescale/linux-fslc/commit/a0d4da5937f3f0c2b8fed8c6bc4c003bea15a34c

Upstream-status: Pending
Signed-off-by: Daiane Angolini <daiane.angolini@foundries.io>
---
 arch/arm64/boot/dts/freescale/imx8mm-evkb.dts | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/freescale/imx8mm-evkb.dts b/arch/arm64/boot/dts/freescale/imx8mm-evkb.dts
index 164df627a213..fe9e19ed498b 100644
--- a/arch/arm64/boot/dts/freescale/imx8mm-evkb.dts
+++ b/arch/arm64/boot/dts/freescale/imx8mm-evkb.dts
@@ -13,7 +13,7 @@ / {
 };

 &i2c1 {
-	/delete-node/ pmic@4b;
+	/delete-node/ pca9450@25;

 	pmic@25 {
 		compatible = "nxp,pca9450a";
--
2.44.1
